tests/integration/sync_test.go

```diff
@@ -7812,3 +7812,237 @@ func Test_Sync_Partials(t *testing.T) {
+			name:             "ca-certificate",
+			initialStateFile: "testdata/sync/039-avoid-entity-rewrite-with-id-on-select-tag-mismatch/ca-certificate-initial.yaml",
+			targetStateFile:  "testdata/sync/039-avoid-entity-rewrite-with-id-on-select-tag-mismatch/ca-certificate-final.yaml",
+			errorExpected:    "error: a CA certificate with ID b824e7c0-d116-4c03-9e27-de05c5d30083 already exists",
+		},
+		{
+			name:             "consumer",
+			initialStateFile: "testdata/sync/039-avoid-entity-rewrite-with-id-on-select-tag-mismatch/consumer-initial.yaml",
+			targetStateFile:  "testdata/sync/039-avoid-entity-rewrite-with-id-on-select-tag-mismatch/consumer-final.yaml",
+			errorExpected:    "error: a consumer with ID 5a1e49a8-2536-41fa-a4e9-605bf218a4fa already exists",
+		},
+		{
+			name:             "plugin",
+			initialStateFile: "testdata/sync/039-avoid-entity-rewrite-with-id-on-select-tag-mismatch/plugin-initial.yaml",
+			targetStateFile:  "testdata/sync/039-avoid-entity-rewrite-with-id-on-select-tag-mismatch/plugin-final.yaml",
+			errorExpected:    "error: a plugin with ID 5a1e49a8-2536-41fa-a4e9-605bf218a4fa already exists",
+		},
+		{
+			name:             "route",
+			initialStateFile: "testdata/sync/039-avoid-entity-rewrite-with-id-on-select-tag-mismatch/route-initial.yaml",
+			targetStateFile:  "testdata/sync/039-avoid-entity-rewrite-with-id-on-select-tag-mismatch/route-final.yaml",
+			errorExpected:    "error: a route with ID 87b6a97e-f3f7-4c47-857a-7464cb9e202b already exists",
+		},
+		{
+			name:             "service",
+			initialStateFile: "testdata/sync/039-avoid-entity-rewrite-with-id-on-select-tag-mismatch/service-initial.yaml",
+			targetStateFile:  "testdata/sync/039-avoid-entity-rewrite-with-id-on-select-tag-mismatch/service-final.yaml",
+			errorExpected:    "error: a service with ID 58076db2-28b6-423b-ba39-a797193017f7 already exists",
+		},
+		{
+			name:             "sni",
+			initialStateFile: "testdata/sync/039-avoid-entity-rewrite-with-id-on-select-tag-mismatch/sni-initial.yaml",
+			targetStateFile:  "testdata/sync/039-avoid-entity-rewrite-with-id-on-select-tag-mismatch/sni-final.yaml",
+			errorExpected:    "error: a SNI with ID 87b6a97e-f3f7-4c47-857a-7464cb9e202b already exists",
+		},
+		{
+			name:             "upstream",
+			initialStateFile: "testdata/sync/039-avoid-entity-rewrite-with-id-on-select-tag-mismatch/upstream-initial.yaml",
+			targetStateFile:  "testdata/sync/039-avoid-entity-rewrite-with-id-on-select-tag-mismatch/upstream-final.yaml",
+			errorExpected:    "error: an upstream with ID 87b6a97e-f3f7-4c47-857a-7464cb9e202b already exists",
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			runWhenKongOrKonnect(t, ">=2.8.0")
+			mustResetKongState(context.TODO(), t, client, deckDump.Config{})
+			err := sync(tc.initialStateFile)
+			require.NoError(t, err)
+
+			err = sync(tc.targetStateFile, "--select-tag", "after")
+			require.ErrorContains(t, err, tc.errorExpected)
```

Comment:
We should "require" for the error here and then "assert" for the error message.
This would stop the test suite if no error comes, so as to avoid panics in next line.
If an assertion fails, the test suite would keep running, which is fine because we don't expect a panic there.

Stopping the entire test suit just because error strings didn't match doesn't make much sense.

```suggestion
			require.Error(t, err)
                        assert.ErrorContains(t, err, tc.errorExpected)
```

Reply:
I'm a bit confused here - we want to use require.Error here as good practice, correct? (here panic is not happening, but it could in other tests if we don't use?)
Also didn't understand `Stopping the entire test suit just because error strings didn't match doesn't make much sense.` - here the rest of the subtests run as expected when when one error string doesn't match, even when require.Error is not used - what am I missing? 🤔 
@Prashansa-K